aflr/api_request.py

```diff
@@ -4,52 +4,72 @@
 import os
 import requests
 from requests.exceptions import HTTPError
+from functools import lru_cache
 
 
 class APIRequest:
-    def __init__(self, api_key=None, api_base=None, **params):
-        self.api_key = api_key if api_key else aflr.api_key
-        self.api_base = api_base if api_base else aflr.api_base
-        if self.api_key == None or self.api_key == "your-key":
-            self.api_key = os.environ.get("aflr_key", None)
-        if self.api_key == None or len(self.api_key) < 30:
+    @lru_cache(maxsize=10, typed=False)
+    def _api_key_checker(api_key=None):
+        if api_key == None or api_key == "your-key":
+            api_key = os.environ.get("aflr_key", None)
+        if not isinstance(api_key, str):
+            raise TypeError("api_key must be of type string.")
+        if api_key == None or len(api_key) < 32:
             raise ValueError(
                 "Please specify a valid api_key or create one here:\nhttps://console.api.audio"
```

Comment:
can we have a space after "\n" in case new line doesnt work

Reply:
sure